src/ServiceModel/ServiceModel.Proxies/ServiceModel/CommunicationObjectExtensions.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.ServiceModel;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace InsightArchitectures.Utilities.ServiceModel
+{
+    /// <summary>
+    /// A set of extension methods around <see cref="ICommunicationObject"/>.
+    /// </summary>
+    public static class CommunicationObjectExtensions
+    {
+        /// <summary>
+        /// Asynchronously opens the connection of the proxy.
+        /// </summary>
+        /// <exception cref="ArgumentNullException">The <paramref name="communicationObject"/> is null.</exception>
+        public static Task OpenAsync(this ICommunicationObject communicationObject)
+        {
+            _ = communicationObject ?? throw new ArgumentNullException(nameof(communicationObject));
+
+            return Task.Factory.FromAsync(communicationObject.BeginOpen(null, null), communicationObject.EndOpen);
+        }
+
+        /// <summary>
+        /// Asynchronously closes the connection of the proxy.
+        /// </summary>
+        /// <exception cref="ArgumentNullException">The <paramref name="communicationObject"/> is null.</exception>
+        public static Task CloseAsync(this ICommunicationObject communicationObject)
+        {
+            _ = communicationObject ?? throw new ArgumentNullException(nameof(communicationObject));
+
+            return Task.Factory.FromAsync(communicationObject.BeginClose(null, null), communicationObject.EndClose);
+        }
+
+        /// <summary>
+        /// Safely disposes the channel by either closing or aborting it.
+        /// </summary>
+        /// <param name="communicationObject">The channel to dispose.</param>
+        /// <param name="logger">A logger to be used for logging purposes.</param>
+        /// <exception cref="ArgumentNullException">The <paramref name="communicationObject"/> is null.</exception>
+        public static async ValueTask DisposeChannelAsync(this ICommunicationObject communicationObject, ILogger? logger = null)
```

Comment:
This method is called twice when running the sample client.
Rightfully so, since the channel is open both times. 
By I wonder why there is more than one connection to dispose 😕

Reply:
Did you see this behavior with both samples?

Reply:
Hm, I must have been mistaken. Just tested both clients now and at the second dispose, the channel state is indeed "closed".
Both tested with NetTcpBindings.

Reply:
Did you create your own test server?

Reply:
Used a PRO WCF app.

Reply:
Oh that's nice :)